crates/runtime/src/dataconnector/graphql/mod.rs

```diff
@@ -0,0 +1,145 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+mod client;
+mod pagination;
+mod table_provider;
+
+use async_trait::async_trait;
+use reqwest::header::{HeaderMap, HeaderValue};
+use reqwest::header::{CONTENT_TYPE, USER_AGENT};
+use secrets::{get_secret_or_param, Secret};
+use url::Url;
+
+use crate::component::dataset::Dataset;
+use datafusion::datasource::TableProvider;
+use snafu::prelude::*;
+use std::any::Any;
+use std::pin::Pin;
+use std::sync::Arc;
+use std::{collections::HashMap, future::Future};
+
+use self::client::{Auth, GraphQLClient};
+use self::pagination::PaginationParameters;
+use self::table_provider::GraphQLTableProvider;
+
+use super::{DataConnector, DataConnectorFactory};
+
+pub struct GraphQL {
+    secret: Option<Secret>,
+    params: Arc<HashMap<String, String>>,
+}
+
+impl DataConnectorFactory for GraphQL {
+    fn create(
+        secret: Option<Secret>,
+        params: Arc<HashMap<String, String>>,
+    ) -> Pin<Box<dyn Future<Output = super::NewDataConnectorResult> + Send>> {
+        Box::pin(async move {
+            let graphql = Self { secret, params };
+            Ok(Arc::new(graphql) as Arc<dyn DataConnector>)
+        })
+    }
+}
+
+impl GraphQL {
+    fn get_client(&self, dataset: &Dataset) -> super::DataConnectorResult<GraphQLClient> {
+        let mut client_builder = reqwest::Client::builder();
+        let token = get_secret_or_param(&self.params, &self.secret, "auth_token_key", "auth_token");
+        let user = get_secret_or_param(&self.params, &self.secret, "auth_user_key", "auth_user");
+        let pass = get_secret_or_param(&self.params, &self.secret, "auth_pass_key", "auth_pass");
+
+        let query = self
+            .params
+            .get("query")
+            .ok_or("`query` not found in params".into())
+            .context(super::InvalidConfigurationSnafu {
+                dataconnector: "GraphQL",
+                message: "`query` not found in params",
+            })?
+            .to_owned();
+        let endpoint = Url::parse(&dataset.path()).map_err(Into::into).context(
+            super::InvalidConfigurationSnafu {
+                dataconnector: "GraphQL",
+                message: "Invalid URL in dataset `from` definition",
+            },
+        )?;
+        let json_path = self
+            .params
+            .get("json_path")
+            .ok_or("`json_path` not found in params".into())
+            .context(super::InvalidConfigurationSnafu {
+                dataconnector: "GraphQL",
+                message: "`json_path` not found in params",
+            })?
+            .to_owned();
+        let pointer = format!("/{}", json_path.replace('.', "/"));
+
+        let pagination_parameters = PaginationParameters::parse(&query, &pointer);
```

Comment:
Why changing from `.` to `/`? Can't we just use `json_path` as is for parsing?

Reply:
Or can we just use that format in spicepod?

Reply:
serde_json uses this format for access
dot notation is more clean for a user